workers/generic-worker/loopback_video_linux_test.go

```diff
@@ -0,0 +1,61 @@
+//go:build linux
+
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/mcuadros/go-defaults"
+)
+
+func TestLoopbackVideo(t *testing.T) {
+	setup(t)
+
+	payload := GenericWorkerPayload{
+		Command: [][]string{
+			{"ls", "-l", "/dev/video0"},
+		},
+		MaxRunTime: 30,
+		Features: FeatureFlags{
+			LoopbackVideo: true,
+		},
+	}
+	defaults.SetDefaults(&payload)
+	td := testTask(t)
+	td.Scopes = append(td.Scopes, "generic-worker:loopback-video")
+
+	_ = submitAndAssert(t, td, payload, "completed", "completed")
+
+	logText := LogText(t)
+	if !strings.Contains(logText, "/dev/video0") {
+		t.Fatalf("Expected log to contain /dev/video0, but it didn't\n%s", logText)
+	}
+	if !strings.Contains(logText, "crw-rw----") {
+		t.Fatalf("Expected log to contain crw-rw----, but it didn't\n%s", logText)
+	}
+}
+
+func TestLoopbackVideoEnvVar(t *testing.T) {
+	setup(t)
+
+	payload := GenericWorkerPayload{
+		Command: [][]string{
+			{"/bin/bash", "-c", "echo $TASKCLUSTER_VIDEO_DEVICE"},
+		},
+		MaxRunTime: 30,
+		Features: FeatureFlags{
+			LoopbackVideo: true,
+		},
+	}
+	defaults.SetDefaults(&payload)
+	td := testTask(t)
+	td.Scopes = append(td.Scopes, "generic-worker:loopback-video")
+
+	_ = submitAndAssert(t, td, payload, "completed", "completed")
+
+	logText := LogText(t)
+	if !strings.Contains(logText, "/dev/video0") {
+		t.Fatalf("Expected log to contain /dev/video0, but it didn't\n%s", logText)
+	}
```

Comment:
I think this should be a regexp, i.e. `/dev/video[0-9]+` (or however you do digits in go regexp)

Reply:
This test should always be device 0, as it's hardcoded in the generic worker test config.

Reply:
Same here, maybe we can read it from config value, so that it works even if test setup is changed to use a different device number. For example, I suspect we can't always assume /dev/video0 is free, if someone is running the tests on a linux development machine that has a webcam, maybe /dev/video0 is already assigned to their real webcam. So I feel like at some point, we should probably change the test setup to use the first available free device, but since none of us are using linux dev machines we probably haven't hit that issue yet. In general, we can't assume /dev/video0 is available.